google/cloud/internal/async_retry_loop.h

```diff
@@ -189,8 +189,12 @@ class AsyncRetryLoopImpl
 
   future<T> Start() {
     auto weak = std::weak_ptr<AsyncRetryLoopImpl>(this->shared_from_this());
-    result_ = promise<T>([weak] {
-      if (auto self = weak.lock()) self->Cancel();
+    auto const& options = CurrentOptions();
```

Comment:
Same comments as in the polling loop.  I think you want to capture these options so they can be set in the context of the callbacks, e.g., when entering (or before calling) `OnAttempt()`, `OnOperation`, or `OnBackoff`.

Reply:
See above.  `OnAttempt()`, `OnOperation()`, and `OnBackoff()` are already called with the loop-start-time options.

Reply:
Thanks